src/types.ts

```diff
@@ -350,17 +351,25 @@ export enum MedicationRequestStatus {
 }
 
 /**
- * This is a bit of a mouthful, but, within the UI, the "status" of a request we want to display to the pharmacist is
- * a combination of the status of the request and the status of the most recent dispensing event; given a request
- * status and a dispense status, this is the actual status we want to display to the pharmacist (see util method
- * computeMedicationRequestMedicationDispenseCombinedStatus)
+ * These fulfiller statuses are an OpenMRS defined status (as opposed to FHIR) that we define as upper-case,
+ * hence the case difference from the other statuses
+ */
+export enum MedicationRequestFulfillerStatus {
+  on_hold = "ON_HOLD",
+  declined = "DECLINED",
+  completed = "COMPLETED",
+}
+
+/**
+ * Within the UI, the "status" of a request we want to display to the pharmacist is
+ * a combination of the status and fulfiller status; this is the actual status we want to display to the pharmacist (see util method
+ * computeMedicationRequestCombinedStatus)
```

Comment:
This is a rework, we since we are now updating the fulfiller status on a medication request, we now calculate the "combined" condition based on the status and fulfiller status of the medication request (and not the medication dispense status).... see comments above.